.npmignore

```diff
@@ -0,0 +1,10 @@
+test
+examples
+.travis.yml
+.gitignore
+.babelrc
+.nyc_output/
+coverage/
+.vscode/
+docs/
+webpack/
```

Comment:
By default, `npm` uses your `.gitignore` to decide what ends up in npm. This is actually usually not what we want - we want source code in `git` and build outputs in `npm`.  Otherwise, end users have to compile the lib _after_ they install, which is not only slower but it makes them prone to Angular's constantly breaking toolchain issues. It also means they have to pull in all our devdeps too.

A better approach is to use a `.npmignore` file to override this; we're putting the build outputs only in `npm`.

That way, instead of doing a `postinstall: npm run build`, we do a `prepublish: npm run build`.